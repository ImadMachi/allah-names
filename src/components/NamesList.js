import React from "react";
import { useSelector } from "react-redux";
import { Link } from "react-router-dom";

const NamesList = () => {
  const names = useSelector((state) => state.names);
  return (
    <div>
      <h1 className="title">
        {"{وَلِلَّهِ الْأَسْمَاءُ الْحُسْنَى فَادْعُوهُ بِهَا}"}
      </h1>
      <span className="names-list">
        {names.map((elem) => (
          <Link to={"/" + elem.id} key={elem.id} className="names-list__name">
            {elem.name}
          </Link>
        ))}
      </span>
    </div>
  );
};

export default NamesList;
