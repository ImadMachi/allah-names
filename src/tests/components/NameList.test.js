import React from "react";
import { shallow } from "enzyme";
import NamesList from "./../../components/NamesList";

test("should render without throwing an error", () => {
  const wrapper = shallow(<NamesList />);
  expect(wrapper).toMatchSnapshot();
});

jest.mock("react-redux", () => ({
  useSelector: jest.fn().mockReturnValue([
    {
      name: "الله",
      description:
        "الاسم الذي يدلّ على ذات الله -تعالى- الجامعة لصفات ألوهيّته.",
      id: "N1",
    },
    {
      name: "الرحمن",
      description:
        "الاسم الذي يدلّ على شمول رحمته كافّة خَلقه؛ بأن خلقهم ورزقهم، وهو اسم يختص به الله -تعالى- ولا يجوز أن يُطلَق على أحد غيره.",
      id: "N2",
    },
    {
      name: "الرحيم",
      description:
        "اسم خاصّ برحمة الله -تعالى- بعباده المؤمنين؛ بهدايتهم للإيمان وإثابتهم الثواب الدائم في الآخرة.",
      id: "N3",
    },
  ]),
}));
