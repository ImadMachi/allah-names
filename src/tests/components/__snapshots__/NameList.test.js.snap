// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without throwing an error 1`] = `
<div>
  <h1
    className="title"
  >
    {وَلِلَّهِ الْأَسْمَاءُ الْحُسْنَى فَادْعُوهُ بِهَا}
  </h1>
  <span
    className="names-list"
  >
    <Link
      className="names-list__name"
      key="N1"
      to="/N1"
    >
      الله
    </Link>
    <Link
      className="names-list__name"
      key="N2"
      to="/N2"
    >
      الرحمن
    </Link>
    <Link
      className="names-list__name"
      key="N3"
      to="/N3"
    >
      الرحيم
    </Link>
  </span>
</div>
`;
